python_ta/transforms/type_inference_visitor.py

```diff
@@ -447,9 +447,11 @@ def visit_functiondef(self, node: astroid.FunctionDef) -> None:
         inferred_args = [self.lookup_type(node, arg) for arg in node.argnames()]
 
         if isinstance(node.parent, astroid.ClassDef) and isinstance(inferred_args[0], TypeVar):
```

Comment:
The first condition may be unnecessary since otherwise node.type won't be 'method' or 'classmethod'. I don't understand the second condition: in what case would looking up an argument not resolve to a TypeVar?